rtsp.go

```diff
@@ -161,8 +162,13 @@ type rtspCamera struct {
 	resource.AlwaysRebuild
 	model resource.Model
 	gostream.VideoReader
-	u *base.URL
+	u             *base.URL
+	lazyDecodeRTP bool
 
+	closeMu sync.RWMutex
```

Comment:
I needed to add this as now both Image and OnPacket's callback (which are called from 2 different goroutines) can try to interact with the pool and the decoder. As a result, I made it so that `Image` can execute concurrently with other `Image` calls but can't run concurrently with `Close` and such that it will early exist if close has already been called.

Reply:
This sounds like it should be implemented with our `StoppableWorker` structure to handle this exact case, can you make a TODO and a ticket to look into this. 

Are you seeing cases where Close is not stopping the Image call? If so there's probably a bug or something being badly freed/uncancelled in this code.

Reply:
It is a similar shape of problem to what we use `StoppableWorker` to solve, however I don't think `StoppableWorker` is the right solution to this problem as the goroutine calling `Image` is spawned by the camera/server.go hosted in the module, not by our code.

I think the approach I used here is reasonable. Close won't do it's thing until all in progress `Image` calls complete and once `Close` is called, `Image` won't execute.